docs/draft/rest_api.md

```diff
@@ -1,201 +1,324 @@
-## General Calls
-### Get current status of the transformer
-```
-GET status/
+# Alternative draft for the REST api
+
+```java
+Note:
+Terms in curly braces like {appName} are meant to be substituted with a proper value. 
+They must not be used literally.
 ```
-**returns:**
-  
-HTTP Response Code: 200 (always)
-  
+![rest api as a tree](rest-api.png)
+### Current status
+##### GET /status
+Get the current status of the transformer.
+*returns:* HTTP Response Code: 200 (always)
+
 ```json
 {
-  "state": "idle",
-  "current_transformations": [1,2,3],
-  "available_storage": 1000,
-  "total_storage": 10000,
-  "capabilities": ["aws", "kubernetes", "openstack"]
+    "status": "idle",
+    "available_storage": 1000,
+    "total_storage": 10000,
 }
 ```
-- *state*: current status of the transformer, values can either be `idle`, `transforming` or `error`
-- *current_transformations:* ids of the active or queued transformations (Array of unsigned integers)
-- *available_storage*: represents the free hard disk space in megabytes (Unsigned integer)
-- *total_storage*: represents the total storage of the hard disk (Unsigned integer)
-- *capabilities*: represents a list of the plugins loaded, i.e. the platforms which are supported by this instance of the transformer
+- *status*: current status of the transformer, values can either be `idle`, `transforming` or `error`
+- *available_storage*: free hard disk storage in MB (integer)
+- *total_storage*: total storage of the hard disk (integer, in MB)
 
-## Transformation specific calls
+### Manage target platforms
 
-***Error Note***: Every Call that requires an ID will return a Error 404 if the ID cannot be found.
+##### GET /platforms
+Returns all target platforms which are available for transforming the CSAR.
 
-### Retrieve the result of the transformation
-
-```
-GET transformation/{id}
-```
-  
-*Parameters*:
-- **id**: The id of the transformation for which the result should be retrieved.
-  
-**on success returns:**
-  
-HTTP Response Code: 200
+*returns:*
 ```json
-{
-  "id": 1,
-  "result": "/results/transform_1.zip"
-}
+[ 
+    "aws": {
+        "href":"/platforms/aws"
+    },
+    "cloudformation": {
+        "href":"/platforms/cloudformation"
+    }
+]
 ```
-- *id*: The id of the transformation (unsigned Integer)
-- *result*: the URL path (on the HTTP server) to download the result archive. If the transformation is still in progress or has errored this value will be null
+NOTE: Later it is possible to add platform specific preferences here
 
-### Getting the status of a transformation
-```
-GET transformation/{id}/status
-```
+### Manipulating TOSCA Models
+
+##### GET /toscamodels
+Get a list of all toscamodels.  
 
-*Parameters*:
-- **id**: The id of the transformation for which the result should be retrieved.
+*returns*:
+```json
+[
+    {
+        "href":"/toscamodels/hello-world",
+        "name":"hello-world",
+        "csar": {
+            "href":"/toscamodels/hello-world/csar"
```

Comment:
Not sure if it is necessary to return the href. It might be easier to use but is in my opinion unnecessary extra information. You can build this href with the name.

But maybe I skipped a information and this common practice.